### PR TITLE
Creates a method to change the font

### PR DESCRIPTION
**Pull request by jnthas**
Hi Dimitris, first of all thanks for that library it's really simple and useful.
I've created a base struct to hold font properties so that I can change the font during execution. It was the only feature missing for me. 
About the supported fonts, it can be any column number (not only five) but the row still needs to be 1 byte. It works for me since I'm needed a smaller one and a bold one. I can share an example if you want.
Just in case you feel it's interesting, I'm opening this pull request.

Tks.

### CODE REVIEW

**Comment by jnthas** [comment]
Example of use
https://gist.github.com/jnthas/81e5021417fed9aee8cdcb3a35956a45
Fonts based on 
https://github.com/gavinlyonsrepo/NOKIA5110_TEXT

File: src/LCD_Fonts.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#pragma once
+#include <Arduino.h>
+
+struct CustomFont {
+    virtual const unsigned char* getFont(char character) = 0;
+    virtual const uint8_t colSize() = 0;
+    virtual const uint8_t rowSize() = 0;
+    virtual const unsigned char* HSpacingChar() = 0;
+    virtual const uint8_t HSpacingSize() = 0;
+};
```
Comment (platisd) [inline src/LCD_Fonts.h]: Unless I am terribly mistaken, we don't need polymorphism here (and if we did need, `CustomFont` should definitely have a virtual destructor). Do you think the following class can accommodate your use case?

```suggestion
class NokiaLcdFont
{
public:
    NokiaLcdFont(const unsigned char fontTable[][Nokia_LCD_Fonts::kColumns_per_character],
                 const unsigned char* hSpace,
                 uint8_t hSpaceSize,
                 uint8_t characterOffset = 0x20)
        : hSpace{ hSpace }
        , hSpaceSize{ hSpaceSize }
        , columnSize{ Nokia_LCD_Fonts::kColumns_per_character }
        , rowSize{ Nokia_LCD_Fonts::kRows_per_character }
        , kFontTable{ fontTable }
        , kCharacterOffset{ characterOffset }
    {
    }
    const unsigned char* getFont(char character) const
    {
        return kFontTable[character - kCharacterOffset];
    }

    const unsigned char* const hSpace;
    const uint8_t hSpaceSize;
    const uint8_t columnSize;
    const uint8_t rowSize;

private:
    const unsigned char (*kFontTable)[Nokia_LCD_Fonts::kColumns_per_character];
    const uint8_t kCharacterOffset;
};
```

File: src/Nokia_LCD.cpp [hunk 2]
```diff
@@ -14,12 +14,17 @@
...
 const uint8_t kDisplay_max_width = 84;
 const uint8_t kDisplay_max_height = 48;
+
+const Nokia_LCD_Fonts::DefaultFont defaultFont;
+
 // Each row is made of 8-bit columns
-const unsigned int kTotal_rows =
-    kDisplay_max_height / Nokia_LCD_Fonts::kRows_per_character;
+unsigned int kTotal_rows =
+    kDisplay_max_height / defaultFont.rowSize();
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: We should not have global variables that are _not_ constants. Before we turn it into a (non-constant) member variable, does it make sense for the total rows to change or be different than the default? Is there a use case for this?

File: src/Nokia_LCD.cpp [hunk 3]
```diff
@@ -14,12 +14,17 @@
...
 const unsigned int kTotal_columns = kDisplay_max_width;
 const unsigned int kTotal_bits = kDisplay_max_width * kTotal_rows;
 const char kNull_char = '\0';
+
+
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: Remove? :cat:

File: src/Nokia_LCD.cpp [hunk 4]
```diff
@@ -86,6 +91,9 @@ void Nokia_LCD::begin() {
         pinMode(kBl_pin, OUTPUT);
     }
 
+    // Sets the default font
+    setFont(&defaultFont);
+
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: We can set the font to the default value in the constructor's initializer list.
E.g. `kUsingHardwareSPI{true}, customFont{&defaultFont}`

File: src/Nokia_LCD.cpp [hunk 5]
```diff
@@ -107,6 +115,15 @@ void Nokia_LCD::setContrast(uint8_t contrast) {
 
 void Nokia_LCD::setInverted(bool invert) { mInverted = invert; }
 
+void Nokia_LCD::setFont(CustomFont *font) { 
+    if (font != NULL) 
+        customFont = font;         
+    else
+        customFont = &defaultFont;
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: Considering the previous comment about setting the font in the constructor's initializer list, this should be turned into:

```suggestion
    if (!font) {
        return;
    }
    customFont = &defaultFont;
```

File: src/Nokia_LCD.cpp [hunk 6]
```diff
@@ -107,6 +115,15 @@ void Nokia_LCD::setContrast(uint8_t contrast) {
...
+    
+    kTotal_rows = kDisplay_max_height / customFont->rowSize();
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: See previous comment about total rows. Do they ever need to change?

File: src/Nokia_LCD_Fonts.h [hunk 7]
```diff
@@ -6,210 +6,238 @@
...
+        {0x00, 0x08, 0x36, 0x41, 0x00}  // 0x7b {
+        ,
+        {0x00, 0x00, 0x7f, 0x00, 0x00}  // 0x7c |
+        ,
+        {0x00, 0x41, 0x36, 0x08, 0x00}  // 0x7d }
+        ,
+        {0x10, 0x08, 0x08, 0x10, 0x08}  // 0x7e ~
+        ,
+        {0x78, 0x46, 0x41, 0x46, 0x78}  // 0x7f DEL
+    };
```
Comment (platisd) [inline src/Nokia_LCD_Fonts.h]: Please preserve the formatting (I.e. undo the indentation)

File: src/Nokia_LCD_Fonts.h [hunk 8]
```diff
@@ -6,210 +6,238 @@
...
+            return kRows_per_character;
+        }
+        
+        const unsigned char* HSpacingChar() {
+            return hSpace;
+        }
+        
+        const uint8_t HSpacingSize() {
+            return 1;
+        }
```
Comment (platisd) [inline src/Nokia_LCD_Fonts.h]: See previous suggestion that avoids inheritance and only requires a user to supply another font table.

**Review by platisd — CHANGES_REQUESTED** [review]
Thanks a lot for your interest to contribute! I think we should get this merged, but before we do, please have look at some comments that I left. :+1:

File: src/LCD_Fonts.h (hunk 1 above)
Comment (jnthas) [reply src/LCD_Fonts.h]: It makes perfect sense! It's so valuable to have another point of view. I will makes the changes and test during the week, I let you know. Thanks for the code review :)

File: src/Nokia_LCD.cpp (hunk 2 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: You're right. We don't have fonts with more than one row height. So, no problem in keeping this fixed to `Nokia_LCD_Fonts::kRows_per_character`

File: src/Nokia_LCD.cpp (hunk 3 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: Done

File: src/Nokia_LCD.cpp (hunk 4 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: Great tip, done

File: src/Nokia_LCD.cpp (hunk 5 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: I got the idea, but just one point here, the last line (122) should be `customFont = font;`, right?

File: src/Nokia_LCD.cpp (hunk 6 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: Got it, removed.

File: src/Nokia_LCD_Fonts.h (hunk 7 above)
Comment (jnthas) [reply src/Nokia_LCD_Fonts.h]: Fixed

File: src/LCD_Fonts.h (hunk 1 above)
Comment (jnthas) [reply src/LCD_Fonts.h]: I get the general idea, using a class to create objects that represent the font is great. But I'm having problems with the font table array. I'm trying to create an LcdFont class, it will have a default constructor that passes Nokia font parameters and another constructor that takes all parameters including a pointer to the font table and that's the problem. Do you have any thoughts on this?

Comment (platisd) [reply src/LCD_Fonts.h]: Not sure what is the problem you are referring to, but why do you need two constructors?

You can use this class as-is and then create a global constant at `src/Nokia_LCD.cpp` like you have already done. E.g.

```cpp
const LcdFont nokiaFont{kDefault_font, space, 1};
```

And then use it pretty much the way you already do in your pull request.

File: src/Nokia_LCD.cpp (hunk 5 above)
Comment (platisd) [reply src/Nokia_LCD.cpp]: Yeap

File: src/LCD_Fonts.h (hunk 1 above)
Comment (jnthas) [reply src/LCD_Fonts.h]: Hi Dimitris, sorry for the delay. I just made it work now. Nevermind the class with two constructors :sweat_smile: I had to do small change in the constructor because the fonts have a different column number and the param that you suggested `const unsigned char fontTable[][Nokia_LCD_Fonts::kColumns_per_character]` won't work.
I also created another method `setDefaultFont()` in order to set the NokiaFont back again ((in case the user is using another font).

Comment (jnthas) [reply src/LCD_Fonts.h]: Hope it is what you expect. I'm open for any improvement anyway.

File: src/Nokia_LCD.cpp [hunk 9]
```diff
@@ -14,6 +14,12 @@
 namespace {
 const uint8_t kDisplay_max_width = 84;
 const uint8_t kDisplay_max_height = 48;
+
+const LcdFont nokiaFont{(const unsigned char*) Nokia_LCD_Fonts::kDefault_font,
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: This cast shouldn't be necessary. If we expect the user to create a 2-D array we should let them pass a 2-D array without decaying it down to a single-dimensional array. :)

File: src/LCD_Fonts.h [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * On how to use this library, please refer to the `examples/` folder.`
+ */
+#pragma once
+#include <Arduino.h>
+#include "Nokia_LCD_Fonts.h"
+class LcdFont
+{
+public:
+    LcdFont(const unsigned char* fontTable,
```
Comment (platisd) [inline src/LCD_Fonts.h]: Is this correct :arrow_right: `const unsigned char* fontTable` ?
We want to pass a 2-dimensional array, but the constructor accepts a single-dimensional one.
Of course we can think of 2-d arrays as 1-d  ones, but since we already represent them in 2-D, why change that?

Have a look at the suggestion I previously sent. That one takes a 2-d array, where the "second" dimension is known at compile time (i.e. `Nokia_LCD_Fonts::kColumns_per_character`) and the class holds a pointer to it (i.e. `const unsigned char (*kFontTable)[Nokia_LCD_Fonts::kColumns_per_character]`).
Is there a reason why the suggestion wouldn't work? On the up side, the user doesn't need to cast the 2-d array to a `const unsigned char*` every time they want to pass it to the constructor.

You can try just the particular class out here: https://onlinegdb.com/3VLNdbM8U

File: src/LCD_Fonts.h (hunk 1 above)
Comment (platisd) [reply src/LCD_Fonts.h]: Made a minor change to the `getFont` method of the comment for more readability.
(In the meanwhile I am preparing my review to your latest patch)

File: src/Nokia_LCD.cpp [hunk 11]
```diff
@@ -47,7 +54,8 @@ Nokia_LCD::Nokia_LCD(const uint8_t dc_pin, const uint8_t ce_pin,
       kUsingBacklight{false},
       kUsingHardwareSPI{true},
       mX_cursor{0},
-      mY_cursor{0} {}
+      mY_cursor{0},
+      currentFont{&nokiaFont} {}
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: Please use the existing naming convention and start member variables with `m`, i.e. `mCurrentFont`

**Review by platisd — CHANGES_REQUESTED** [review]
My main issue is with the LcdFont class and it taking a single-dimensional array as explain in the relevant comment.

Oh also, would you mind adding your example (you had a link to a gist somewhere) to the `examples` directory? To illustrate how one would add their own font.

File: src/Nokia_LCD.cpp (hunk 11 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: Sure! By the way, you usually prefix private members with "k". I'm not an expert of C++, do you have some reference talking about these conventions?

File: src/LCD_Fonts.h (hunk 10 above)
Comment (jnthas) [reply src/LCD_Fonts.h]: Okay, let me try to explain, maybe it could be a lack of understanding on my part.

There are two premises that are guiding me, it's better to make them clear here:
- Custom fonts are not inside the library, they are in the project that is using it. So I need a common class for any font type.
- As you suggested, I'm avoiding using inheritance, just a simple class where the objects instantiated are fonts of different sizes.

In your example, NokiaLcdFont works great but only for 5x8 fonts
`NokiaLcdFont f{ fontTable, hSpace, 1, 0 };`

I have a font called SmallFont with 3x8 and another, bold with 7x8. I could create the classes SmallLcdFont and BoldLcdFont respectively, like you did with NokiaLcdFont. Bbut I wouldn't be able to pass them as a parameter to the library, since they are different classes and don't have a common parent.

`Nokia_LCD::setFont(const ??? *font)` <- ??? what type here?

So I created an LcdFont class that receives fonts of any size and that's why the `const unsigned char* fontTable` was used. If you have a better solution, please let me know. I'm not an C++ expert and I maybe there is something that solves the problem gracefully.

Comment (jnthas) [reply src/LCD_Fonts.h]: I updated the gist https://gist.github.com/jnthas/81e5021417fed9aee8cdcb3a35956a45.
Think that the class example.cpp is outside of the library and just importing it.

Comment (platisd) [reply src/LCD_Fonts.h]: > In your example, NokiaLcdFont works great but only for 5x8 fonts

OK, I see what you mean. The class I suggested indeed accepts `5xColumns` fonts only. Let me think of it and I will get back to you soon.

File: src/Nokia_LCD.cpp (hunk 11 above)
Comment (platisd) [reply src/Nokia_LCD.cpp]: Actually it is the **constants** I prefix with `k`. Variables get the `m` prefix. So `currentFont` should become `mCurrentFont` not `mcurrentFont`. :smiley:

Comment (platisd) [reply src/Nokia_LCD.cpp]: The `m` prefix for member variables is relatively common in different languages. I don't think I follow any particular style (e.g. the Google C++ Style) for this repository.

File: src/LCD_Fonts.h (hunk 10 above)
Comment (platisd) [reply src/LCD_Fonts.h]: OK, there are different ways we can adopt to solve the problem:
1. **Inheritance** like you initially suggested :arrow_right: This isn't a bad way to do things to be honest, but I fear there may make the size of the executable larger which is often an issue on ATtiny microcontrollers. The next drawback was that a user would have to create their own class which may or not be easy.
2. **Templatize** the `LcdFont` class :arrow_right: This would allow us to support pretty much all shapes and sizes. The problem is that we would need to do a large refactoring and turn the other classes into templates too. The change would be backward incompatible. 
3. Let the user bother with the array. This is similar to the first solution where we let the user define their own class and methods with the difference that the user wouldn't need to create their own class, but only a single function. I am currently leaning towards this, but let me show you how it would look like:

```cpp
using GetFontCallback = const unsigned char* (*)(char);

class LcdFont
{
public:
    LcdFont(GetFontCallback getFontCallback,
            uint8_t columnSize,
            const unsigned char* hSpace,
            uint8_t hSpaceSize,
            uint8_t rowSize = 8)
        : hSpace{ hSpace }
        , hSpaceSize{ hSpaceSize }
        , columnSize{ columnSize }
        , rowSize{ rowSize }
        , mGetFontCallback{ getFontCallback }
    {
    }
    const unsigned char* getFont(char character) const
    {
        // We could have the offset here perhaps instead? I don't remember why we need it.
        return mGetFontCallback(character);
    }
    const unsigned char* const hSpace;
    const uint8_t hSpaceSize;
    const uint8_t columnSize;
    const uint8_t rowSize;

private:
    const GetFontCallback mGetFontCallback;
};
```

And a user would use it like this:

```cpp
const unsigned char fontTable[][3]{ { 1, 2, 3 }, { 4, 5, 6 }, { 7, 8, 9 } };
unsigned char hSpace[]{ 0 };
LcdFont f{ [](char c) { return fontTable[c - 0x20]; }, 3, hSpace, 1 };
```
Or if someone doesn't like lambdas, they alternatively could do:

```cpp
const unsigned char fontTable[][3]{ { 1, 2, 3 }, { 4, 5, 6 }, { 7, 8, 9 } };
unsigned char hSpace[]{ 0 };
const unsigned char* getFont(char character)
{
    return fontTable[character - 0x20];
}
LcdFont f{ getFont, 3, hSpace, 1 };
```

Let me know if you understand the solution and what you think of it. :+1:

Comment (jnthas) [reply src/LCD_Fonts.h]: Ok, I got it. Let me change the code and see what happens. 
About the character offset, we should keep outside like you did, that way the users can have a full font table with 256 characters if they want.
And there is another thing that I noticed here now, we could remove rowSize from the constructor, it's optional there but it actually do not, the characters must be 8-bit size always.

Comment (jnthas) [reply src/LCD_Fonts.h]: I like the idea very much, thanks for the suggestion. I just push the changes. I have two questions:
- I could instantiate the LcdFont object (nokiaFont) inside the `Nokia_LCD_Fonts` instead of the namespace of the main class, what do you think?
-  About the example folder, is it worth to create another example to demonstrate how to use a different font?

File: src/LCD_Fonts.h [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+ * It is a derivative of Jim Lindblom's work, from Sparkfun. The goal
+ * was to decrease its size so it can be ran on ATTiny microcontrollers
+ * that have limited RAM size. Therefore, the main difference is that a
+ * framebuffer, which would consume all dynamic memory, is not used. Get the
+ * original library [here](https://github.com/sparkfun/GraphicLCD_Nokia_5110).
+ *
+ * On how to use this library, please refer to the `examples/` folder.`
+ */
+#pragma once
+#include <Arduino.h>
```
Comment (platisd) [inline src/LCD_Fonts.h]: Including `Arduino.h` is actually not necessary. This is pure C++. :}

File: src/Nokia_LCD.cpp [hunk 13]
```diff
@@ -14,9 +14,18 @@
...
 const uint8_t kDisplay_max_width = 84;
 const uint8_t kDisplay_max_height = 48;
+
+// Instantiate the default font
+const LcdFont nokiaFont { 
+    [](char c) { return Nokia_LCD_Fonts::kDefault_font[c - 0x20]; }, 
+    Nokia_LCD_Fonts::kColumns_per_character,
+    Nokia_LCD_Fonts::hSpace, 
+    1 
+};
```
Comment (platisd) [inline src/Nokia_LCD.cpp:24]: :+1:

File: src/Nokia_LCD.cpp [hunk 14]
```diff
@@ -14,9 +14,18 @@
...
+
 // Each row is made of 8-bit columns
 const unsigned int kTotal_rows =
-    kDisplay_max_height / Nokia_LCD_Fonts::kRows_per_character;
+    kDisplay_max_height / nokiaFont.rowSize;
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: Hmmm, is this correct with a different font, i.e. a smaller one? I guess if we use a smaller font then we could have more rows?
Should this perhaps be part of the `LcdFont` class instead? Then the way we calculate `kTotal_bits` may be reconsidered.

I may not have thought this through completely. :thinking:

**Review by platisd — CHANGES_REQUESTED** [review]
Looking better and better. :+1: 

I've left some remarks and regarding your comment, yes, I think an example should be added to illustrate how one can change the font to a custom one.

File: src/LCD_Fonts.h (hunk 10 above)
Comment (platisd) [reply src/LCD_Fonts.h]: > I could instantiate the LcdFont object (nokiaFont) inside the Nokia_LCD_Fonts instead of the namespace of the main class, what do you think?

Yeap, looking at your change you did it exactly like you should. :+1: 

> About the example folder, is it worth to create another example to demonstrate how to use a different font?

Absolutely! :100:

File: src/Nokia_LCD.cpp (hunk 14 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: It's correct, because even the small font uses 8 bit height (remember rowSize = 8). Of course, it's possible to use less (or more) bits than 8 for a single row like the Adafruit implementation, but it involves a more complex refactoring on the `printCharacter()` and other methods. So, in short, all fonts have same height and different width. 

> Should this perhaps be part of the LcdFont class instead?

I though the same while coding, but I need a static constant there and use it like `LcdFont::ROWS_PER_CHARACTER`. If you feel ok with this, I can change it.

IMG_20210724_151815

File: src/LCD_Fonts.h (hunk 12 above)
Comment (jnthas) [reply src/LCD_Fonts.h]: Removed!

File: src/Nokia_LCD.cpp (hunk 14 above)
Comment (platisd) [reply src/Nokia_LCD.cpp]: Yes, let's leave it as it was and use the constant here so not to give the false impression that this depends on the default font.

**Review by platisd — CHANGES_REQUESTED** [review]
Three changes:
* Use the static constant as discussed
* Add a new example on how to use `setFont`
* Bump the library version to `2.5.0` in [library.properties](https://github.com/platisd/nokia-5110-lcd-library/blob/master/library.properties) so that it can be ready for release

**Comment by jnthas** [comment]
I've included three font headers in the example folder in order to demonstrate how to use it. Let me know if the example is ok or  needs to change something. I also bump the version and created the rowSize static constant.

File: examples/Nokia_LCD_Demo/fonts/Glyphs_LCD_Fonts.h [hunk 15]
```diff
@@ -0,0 +1,294 @@
...
+    //
+    // const unsigned char* getFont(char character)
+    // {
+    //     return kFont_Table[character];
+    // }
+
+    const LcdFont GlyphFont { 
+        [](char c) { return kFont_Table[(uint8_t) c]; }, 
+        kColumns_per_character,
+        NULL,                       // This font has no aditional horizontal spacing
```
Comment (platisd) [inline examples/Nokia_LCD_Demo/fonts/Glyphs_LCD_Fonts.h]: Please use `nullptr` instead, `NULL` is for `C` :)

File: src/LCD_Fonts.h [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+ * On how to use this library, please refer to the `examples/` folder.`
+ */
+#pragma once
+
+using GetFontCallback = const unsigned char* (*)(char);
+
+class LcdFont
+{
+public:
+    const static uint8_t ROWS_PER_CHARACTER = 8;
```
Comment (platisd) [inline src/LCD_Fonts.h]: I'd say remove this field from here completely and use `Nokia_LCD_Fonts::kRows_per_character` instead that already exists.

File: src/Nokia_LCD.cpp [hunk 17]
```diff
@@ -14,9 +14,18 @@
...
+    [](char c) { return Nokia_LCD_Fonts::kDefault_font[c - 0x20]; }, 
+    Nokia_LCD_Fonts::kColumns_per_character,
+    Nokia_LCD_Fonts::hSpace, 
+    1 
+};
+
 // Each row is made of 8-bit columns
 const unsigned int kTotal_rows =
-    kDisplay_max_height / Nokia_LCD_Fonts::kRows_per_character;
+    kDisplay_max_height / LcdFont::ROWS_PER_CHARACTER;
```
Comment (platisd) [inline src/Nokia_LCD.cpp]: As explained above, basically just undo this change and remove the added static field.

**Review by platisd — CHANGES_REQUESTED** [review]
Two minor changes left and we are ready to merge! :smile: :medal_military: 

Also, did you create these fonts by yourself? If not, where did you get them from? We may need to look into how those are licensed and attribute the original author.

File: examples/Nokia_LCD_Demo/fonts/Glyphs_LCD_Fonts.h (hunk 15 above)
Comment (jnthas) [reply examples/Nokia_LCD_Demo/fonts/Glyphs_LCD_Fonts.h]: Done

File: src/LCD_Fonts.h (hunk 16 above)
Comment (jnthas) [reply src/LCD_Fonts.h]: Oh, okay. I got it

File: src/Nokia_LCD.cpp (hunk 17 above)
Comment (jnthas) [reply src/Nokia_LCD.cpp]: Done

File: examples/Nokia_LCD_Demo/Nokia_LCD_Font.ino [hunk 18]
```diff
@@ -0,0 +1,78 @@
+#include <Nokia_LCD.h>
```
Comment (platisd) [inline examples/Nokia_LCD_Demo/Nokia_LCD_Font.ino]: As per Arduino convention (the file does not compile otherwise) a sketch file should have the same name as the directory. So I suggest changing the directory name to `Nokia_LCD_Font` to match the name of this sketch.

**Review by platisd — CHANGES_REQUESTED** [review]
I added a GitHub Action to automatically build the library and the examples. And it was useful because [it found a mistake](https://github.com/platisd/nokia-5110-lcd-library/actions/runs/1064900830) with the name of the file/directory! Please address that as well. :)

File: src/Nokia_LCD.cpp (hunk 17 above)
Comment (platisd) [reply src/Nokia_LCD.cpp]: Did you forget to push? :scream:

**Review by platisd — CHANGES_REQUESTED** [review]
There's an issue with the licensing of the additional fonts. They are published under LGPL v2.1 while the rest of this library is under MIT. These two licenses are not compatible so we need to use a separate license for the example sketch.

Can you please also add [a file containing this license](https://github.com/risitt/nokia5110Flex/blob/master/LICENSE)  in the same directory as the sketch?

File: examples/Nokia_LCD_Demo/Nokia_LCD_Font.ino (hunk 18 above)
Comment (jnthas) [reply examples/Nokia_LCD_Demo/Nokia_LCD_Font.ino]: I see, I did the change. Wise decision to add this validation :D

**Review by platisd — APPROVED** [review]
Looks good, great job! :100: :1st_place_medal:
Thanks for tolerating my picky comments! :grin:

**Comment by jnthas** [comment]
Thanks, Dimitris. Don't worry with the comments, it was a great opportunity for me to learn a lot of things.
